EIPS/eip-BN256_HashToCurve.md

```diff
@@ -0,0 +1,342 @@
+---
+eip: TBD
+title: Precompile for BN256 HashToCurve Algorithms
+author: Dr. Christopher Gorman, MadHive (@chgormanMH)
+discussions-to: https://ethereum-magicians.org/t/pre-compile-for-bls/3973
+status: Draft
+type: Standards Track
+category: Core
+created: 2020-10-23
+requires: EIP-198, EIP-1108
+---
+
+## Simple Summary
+
+This EIP opens up opportunities for greater use of BN256
+elliptic curve operations by adding a precompiled contract for a
+deterministic hash-to-curve algorithm.
+In particular, this enables BLS signature verification in the EVM
+and reduces its cost to essentially that of one pairing check.
+
+There is a similar proposal to add precompiled contracts for the
+elliptic curve BLS12-381 in
+[EIP-2537](./eip-2537.md);
+this proposal includes a deterministic map-to-curve algorithm.
+
+## Abstract
+There is currently no inexpensive way to perform BLS signature
+verification for arbitrary messages.
+This stems from the fact that there is no precompiled contract
+in the EVM for a hash-to-curve algorithm for the BN256 elliptic curve.
+The gas cost of calling a deterministic hash-to-curve algorithm
+written in Solidity is approximately that of one pairing check,
+although the latter requires an order of magnitude
+more computation.
+This EIP seeks to remedy this by implementing an algorithm
+for hashing to BN256 G1 curve, which would reduce the cost of
+signature verification to essentially that of the pairing check
+precompiled contract.
+We also include a hash-to-curve algorithm for the BN256 G2 group.
+
+## Motivation
+
+The precompiled contracts in
+[EIP-198](./eip-198.md) and
+[EIP-1108](./eip-1108.md)
+increased usage of cryptographic operations in the EVM
+by reducing the gas costs.
+In particular, the cost reduction from
+[EIP-1108](./eip-1108.md)
+helps increase the use of SNARKs in Ethereum
+via an elliptic curve pairing check;
+however, a hash-to-curve algorithm enabling arbitrary
+BLS signature verification on BN256 in the EVM was noticeably missing.
+There is interest in having a precompiled contract which would allow
+for signature verification, as noted
+[here](https://ethereum-magicians.org/t/pre-compile-for-bls/3973).
+If added, [EIP-2537](./eip-2537.md)
+would enable BLS signature verification on the BLS12-381 elliptic curve;
+however, no precompiled contracts currently exist
+for BLS12-381 elliptic curve operations.
```

Comment:
By referring to 2537 inside this EIP you are creating a dependency on 2537 being final before this can go final.  I recommend removing that dependency, since you are explicitly wanting this EIP to be an alternative to 2537 IIUC.  When authoring EIPs, I recommend authoring them with a future reader in mind, rather than a historic reader.  Imagine in 5 or 10 years when someone is reading this EIP after it has been implemented and 2537 was not whether they are going to care at all about 2537.

Reply:
I have removed all references to EIP-2537 from the EIP.